pkg/gardenadm/botanist/etcd.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package botanist
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	sharedcomponent "github.com/gardener/gardener/pkg/component/shared"
+	gardenletconfigv1alpha1 "github.com/gardener/gardener/pkg/gardenlet/apis/config/v1alpha1"
+	imagevectorutils "github.com/gardener/gardener/pkg/utils/imagevector"
+)
+
+// DeployEtcdDruid deploys the etcd-druid component.
+func (b *AutonomousBotanist) DeployEtcdDruid(ctx context.Context) error {
+	var componentImageVectors imagevectorutils.ComponentImageVectors
+	if path := os.Getenv(imagevectorutils.ComponentOverrideEnv); path != "" {
```

Comment:
Why do we use this special image vector override handling? Should we not implement the image vector override in general?

Reply:
This is nothing new but the same as in the seed controller: https://github.com/gardener/gardener/blob/0cf9b8e4619901de2c2a2ffab3b6e1d1208920f7/pkg/gardenlet/controller/seed/add.go#L39-L44